### PR TITLE
chore: Add "client-contrib" as new repo

### PR DESCRIPTION
**Pull request by rhuss**

### CODE REVIEW

File: config/prow/issue_tracker_config.go [hunk 1]
```diff
@@ -69,6 +69,15 @@ func generateIssueTrackerPeriodicJobs() {
 		daysToRot:   rotDefault,
 		daysToClose: closeDefault,
 	}.generateJobs()
+
+	// generate for knative/client-contrib
```
Comment (chizhg) [inline config/prow/issue_tracker_config.go]: Do you really need it now for this new repo? It's not enabled for `knative/client` yet 😆

File: config/prow/config_knative.yaml [hunk 2]
```diff
@@ -151,6 +151,18 @@ presubmits:
         - "./test/presubmit-integration-tests-latest-release.sh"
       dot-dev: true
 
+  knative/client-contrib:
+    - repo-settings:
+      go112-branches: []
```
Comment (chizhg) [inline config/prow/config_knative.yaml]: For now we'll have to add a dummy branch here to make the config-generator work :( 
See https://github.com/knative/test-infra/blob/810ad47815885a5f2ceee52539d4aef4ec4486d5/config/prow/config_knative.yaml#L287

File: config/prow/config_knative.yaml [hunk 3]
```diff
@@ -151,6 +151,18 @@ presubmits:
         - "./test/presubmit-integration-tests-latest-release.sh"
       dot-dev: true
 
+  knative/client-contrib:
```
Comment (chizhg) [inline config/prow/config_knative.yaml]: To be consistent, it's better to also add the simple periodics part for this client-contrib repo, see the setting for client - https://github.com/knative/test-infra/blob/810ad47815885a5f2ceee52539d4aef4ec4486d5/config/prow/config_knative.yaml#L444

File: config/prow/issue_tracker_config.go (hunk 1 above)
Comment (rhuss) [reply config/prow/issue_tracker_config.go]: ok, we can skip that for now (and I will discuss with the team whether we want to enable the stale bot for all client stuff)

File: config/prow_setup.md [hunk 4]
```diff
@@ -69,7 +69,7 @@
 1. Make sure that _Knative Robots_ is an Admin of the repo.
 
 1. Add the new repo to [config_knative.yaml](./prow/config_knative.yaml),
-   without any presubmits. Run `make config` to regenerate
+   without any presubmits. Check the top-level section `presubmits:` and `periodics:` for blueprints for what to add. Then run `make config` to regenerate
```
Comment (rhuss) [inline config/prow_setup.md]: @chizhg I'm not sure what is meant "without any presubmits". Actually you _have to_ add a new section to `presubmits:`, or ? I was a bit confused what to add initially, so I just tried to copy over the config for `knative/client` (but overlooked the periodics part). A bit clarification (maybe even with an example ?) would be cool here.

Also, a link to reference documentation for the possible fields would be awesome (if such documentation exists).

Comment (chizhg) [reply config/prow_setup.md]: You can ignore that "without any presubmits" part. The doc is assuming you will create two PRs, one is setting up Prow for a new repo, and the next one is adding Prow job for the repo, but one PR that contains both is also fine.

Currently we don't have other reference doc for this, but thanks for helping us identify this gap, we will definitely make some improvements for the docs.

**Comment by chizhg** [comment]
Please run `make config` to regenerate the config files.

File: config/prow/config_knative.yaml [hunk 5]
```diff
@@ -471,6 +485,20 @@ periodics:
       dot-dev: true
       go113: true
 
+  knative/client-contrib:
+    - continuous: true
+      dot-dev: true
+      go113: true
+    - nightly: true
```
Comment (chizhg) [inline config/prow/config_knative.yaml]: You haven't added `./hack/release.sh` in that repo, so these jobs (nightly, dot-release, auto-release)  are not needed.

Comment (rhuss) [reply config/prow/config_knative.yaml]: good point. I will remove it for now and add later if needed.

**Pull request merged**